x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/operators/convert_inference_events_to_streaming_events.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { OperatorFunction, map } from 'rxjs';
+import { v4 } from 'uuid';
+import {
+  ChatCompletionEvent as InferenceChatCompletionEvent,
+  ChatCompletionEventType as InferenceChatCompletionEventType,
+} from '@kbn/inference-common';
+import {
+  ChatCompletionChunkEvent,
+  TokenCountEvent,
+  ChatCompletionMessageEvent,
+  StreamingChatResponseEventType,
+} from '../../../../common';
+
+export function convertInferenceEventsToStreamingEvents<
+  T extends InferenceChatCompletionEvent
+>(): OperatorFunction<T, ChatCompletionChunkEvent | TokenCountEvent | ChatCompletionMessageEvent> {
```

Comment:
Does this function have to be generic? Would this work too?

```suggestion
export function convertInferenceEventsToStreamingEvents(): OperatorFunction<
  InferenceChatCompletionEvent,
  ChatCompletionChunkEvent | TokenCountEvent | ChatCompletionMessageEvent
> {
```